crates/primitives/src/transaction/mod.rs

```diff
@@ -1368,13 +1393,13 @@ impl TransactionSigned {
         };
 
         #[cfg(not(feature = "optimism"))]
-        let signature = Signature::decode(data)?;
+        let signature = Signature::decode_rlp_vrs(data)?;
 
         #[cfg(feature = "optimism")]
         let signature = if tx_type == TxType::Deposit {
-            Signature::optimism_deposit_tx_signature()
+            optimism_deposit_tx_signature()
         } else {
-            Signature::decode(data)?
+            Signature::decode_rlp_vrs(data)?
```

Comment:
this should also do additional validation to ensure that `v` was encoded as 0 or 1. earlier this happened on its own because we had a bool there, but needs to be checked explicitly now